baselines/fedht/fedht/main.py

```diff
@@ -0,0 +1,164 @@
+        Config file for federated baseline; read from fedht/conf.
+    """
+    # set seed
+    random.seed(2024)
+
+    # this vs. setting in cfg; what is preferred?
+    device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+
+    if cfg.data == "mnist":
+        # set up mnist data
+        # set partitioner
+        partitioner = PathologicalPartitioner(
+            num_partitions=cfg.num_clients,
+            partition_by="label",
+            num_classes_per_partition=2,
+            class_assignment_mode="first-deterministic",
+        )
+
+        # load MNIST data
+        num_features = 28 * 28
+        num_classes = 10
+        dataset = FederatedDataset(dataset="mnist", partitioners={"train": partitioner})
+        test_dataset = dataset.load_split("test").with_format("numpy")
+        testloader = DataLoader(test_dataset, batch_size=cfg.batch_size, shuffle=False)
+
+        # define model
+        model = LogisticRegression(num_features, num_classes)
+
+        # set client function
+        client_fn = generate_client_fn_mnist(
+            dataset,
+            num_features=num_features,
+            num_classes=num_classes,
+            model=model,
+            cfg=cfg,
+            device=device
+        )
+
+    elif cfg.data == "simII":
+
+        # simulate data from Simulation II in Tong et al
+        num_obs = 1000
+        num_clients = cfg.num_clients
+        num_features = 1000
+        num_classes = 2
+
+        dataset = sim_data(num_obs, num_clients, num_features, 1, 1)
+        X_test, y_test = sim_data(num_obs, 1, num_features, 1, 1)
+        test_dataset = MyDataset(X_test[0, :, :], y_test[:, 0])
+        testloader = DataLoader(test_dataset, batch_size=cfg.batch_size, shuffle=False)
+
+        # define model
+        model = LogisticRegression(num_features, num_classes)
+
+        # set client function
+        client_fn = generate_client_fn_simII(
+            dataset,
+            num_features=num_features,
+            num_classes=num_classes,
+            model=model,
+            cfg=cfg,
+            device=device
+        )
+
+    # initialize global model to all zeros
+    weights = np.zeros((num_classes, num_features))
+    bias = np.zeros(num_classes)
+    init_params_arr: NDArrays = [weights, bias]
+    init_params = ndarrays_to_parameters(init_params_arr)
+
+    # define strategy: fedht
+    strategy_fedht = FedHT(
+        min_available_clients=cfg.strategy.min_available_clients,
+        num_keep=cfg.num_keep,
+        evaluate_fn=get_evaluate_fn(testloader, model, device),
+        on_fit_config_fn=fit_round,
+        iterht=cfg.iterht,
+        initial_parameters=init_params,
+    )
+
+    # define strategy: fedavg
+    strategy_fedavg = fl.server.strategy.FedAvg(
+        min_available_clients=cfg.strategy.min_available_clients,
+        evaluate_fn=get_evaluate_fn(testloader, model, device),
+        on_fit_config_fn=fit_round,
+        initial_parameters=init_params,
+    )
```

Comment:
Could we put these as part of the `if/else` below?

Reply:
added into if/else statement below